spec/models/endpoint_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Endpoint, type: :model do
+  context 'when created' do
+    subject { described_class.new }
+
+    it 'requires a kimono_id' do
```

Comment:
I can see value in keeping it separate, and this PR just depends on that one.  But ultimately it is your call.